capstone/src/main/java/com/google/sps/data/Lesson.java

```diff
@@ -95,6 +102,33 @@ public Date getDate() {
         return (Date) this.entity.getProperty(Lesson.DATE_PROPERTY_KEY);
     }
 
+    public static Lesson deserializeJson(String json) throws IllegalArgumentException {
+        JsonObject jobject = JsonParser.parseString(json).getAsJsonObject();
+        
+        switch(jobject.get(Lesson.TYPE_PROPERTY_KEY).getAsString()) {
```

Comment:
This throws a NullPointerException (when there is no field for type) which is caught in my tests. I'm thinking a more informative error is IllegalArgumentException and so I can do:
String type;
try {
    type = jobject.get(Lesson.TYPE_PROPERTY_KEY).getAsString());
} catch (NullPointerException e){
    throw new IllegalArgumentException();
}

Or is this not necessary since we are catching an unchecked exception?

Reply:
Yep, thats exactly what I would do